test/libsolidity/semanticTests/storage/storage_boundary_struct_array_mixed_types.sol

```diff
@@ -0,0 +1,169 @@
+        // Starts at -20, ends at slot 19
+        assembly {
+            arr.slot := sub(0, 20)
+        }
+    }
+
+    function getDest() internal pure returns (S[10][1] storage arr) {
+        assembly {
+            arr.slot := 21
+        }
+    }
+
+    function getNeighbor() internal pure returns (Neighbor storage n) {
+        // Array ends at slot 19, neighbor at slot 20
+        assembly {
+            n.slot := 20
+        }
+    }
+
+    constructor() {
+        getNeighbor().value = 0x42424242;
+    }
+
+    function fillBoundaryArray() public {
+        S[10][1] storage arr = getBoundaryArray();
+        for (uint i = 0; i < 10; i++) {
+            arr[0][i] = S({
+                a: 1 + i * 5,
+                b: uint128(2 + i * 5),
+                c: uint64(3 + i * 5),
+                d: bytes32(uint256(4 + i * 5)),
+                e: true
+            });
+        }
+    }
+
+    function deleteBoundaryArray() public {
+        S[10][1] storage arr = getBoundaryArray();
+        delete arr[0];
+    }
+
+    function copyFromBoundary() public {
+        S[10][1] storage source = getBoundaryArray();
+        S[10][1] storage dest = getDest();
+        dest[0] = source[0];
+    }
+
+    function copyToBoundary() public {
+        S[10][1] storage source = getDest();
+        S[10][1] storage dest = getBoundaryArray();
+        dest[0] = source[0];
+    }
+
+    function fillDestArray() public {
+        S[10][1] storage dest = getDest();
+        for (uint i = 0; i < 10; i++) {
+            dest[0][i] = S({
+                a: 51 + i * 5,
+                b: uint128(52 + i * 5),
+                c: uint64(53 + i * 5),
+                d: bytes32(uint256(54 + i * 5)),
+                e: true
+            });
+        }
+    }
+
+    function boundaryArray() public view returns (S[10] memory) {
+        return getBoundaryArray()[0];
+    }
+
+    function destArray() public view returns (S[10] memory) {
+        return getDest()[0];
+    }
+
+    function neighborValue() public view returns (uint256) {
+        return getNeighbor().value;
+    }
+}
+// ----
+// neighborValue() -> 0x42424242
+// boundaryArray() -> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+// gas irOptimized: 113161
+// gas legacy: 120742
+// gas legacyOptimized: 112518
+// destArray() -> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+// gas irOptimized: 113048
+// gas legacy: 120716
+// gas legacyOptimized: 112483
+// fillBoundaryArray()
+// gas irOptimized: 912514
+// gas legacy: 930728
+// gas legacyOptimized: 916628
+// neighborValue() -> 0x42424242
+// boundaryArray() -> 1, 2, 3, 4, 1, 6, 7, 8, 9, 1, 11, 12, 13, 14, 1, 16, 17, 18, 19, 1, 21, 22, 23, 24, 1, 26, 27, 28, 29, 1, 31, 32, 33, 34, 1, 36, 37, 38, 39, 1, 41, 42, 43, 44, 1, 46, 47, 48, 49, 1
```

Comment:
You can use `true` and `false` in expectations. Would be helpful here, because when I noticed the ones at first I thought it was a bug. But it's just that this test is constructed differently :)

```suggestion
// boundaryArray() -> 1, 2, 3, 4, true, 6, 7, 8, 9, true, 11, 12, 13, 14, true, 16, 17, 18, 19, true, 21, 22, 23, 24, true, 26, 27, 28, 29, true, 31, 32, 33, 34, true, 36, 37, 38, 39, true, 41, 42, 43, 44, true, 46, 47, 48, 49, true
```